build_mt.py

```diff
@@ -1,599 +1,827 @@
+#!/usr/bin/env python3
+'''
+This is the first script in the "lightcone halo" pipeline.  The goal of this script is to use merger
+tree information to flag halos that intersect the lightcone and make a unique determination of which
+halo catalog epoch from which to draw the halo.
+
+Usage
+-----
+$ ./build_mt.py --help
+'''
+
+import sys
 import glob
+import time
+import gc
+import os
+from pathlib import Path
+
 import asdf
 import numpy as np
-import sys
 from scipy.interpolate import interp1d
 import matplotlib.pyplot as plt
-import numpy.lib.recfunctions as rfn
-import time
-import gc
-import os
+import numba as nb
 
 from tools.merger import simple_load, get_slab_halo, extract_superslab
 
+# these are probably just for testing; should be removed for production
+DEFAULTS = {}
+DEFAULTS['sim_name'] = "AbacusSummit_highbase_c000_ph100"  # AbacusSummit_base_c000_ph006
+DEFAULTS['z_start'] = 0.3  # 0.8#0.5
+DEFAULTS['z_stop'] = 0.65  # 1.25#0.8#0.5
+
+
```

Comment:
I like making any defaults/constants explicit at the top of the file.  For testing, these are probably fine defaults so you don't have to save the command line invocation with all of them. But eventually, these might go away.